### ISSUE TITLE
App stalls upon map tap

### ISSUE THREAD

**Issue by sonora** [osmandapp/OsmAnd#19193]
I've noticed for quite a few weeks now that if you show the map screen at an overview zoom (perhaps 6 or 7) and you tap an area where a big city is (perhaps you try to tap on a favorite you have there), the app stalls, the map screen freezes for some time (could be a minute or so), and often Android warns about a stalled app and offers to close it or wait.

I believe the issue is caused by too many items the app sifts through to compile the selection menu eventually popping up asking what you meant to tap. I believe we filtered that list better before, depending on the zoom level the map is at when being tapped. On some zoom levels I now even seem to receive hiking trails or similar routes in the selection dialog, even though I tapped the map in the map browsing profile.

**Comment by yuriiurshuliak** [comment]
The bug has been reproduced. Try adding a favorite and scaling the map down to 7.
OsmAnd~ 4.7.0#1691m, released: 2024-02-27

https://github.com/osmandapp/OsmAnd/assets/127092082/2abe1e50-4106-474b-a6c3-e84e65248fcc

**Comment by sonora** [comment]
@yuriiurshuliak Yes, thank you. The bug I am talking abut is really from 0:24 on your video, when you finally tapped the map and things then stall. (Before that there is no issue in your video, I think.)

**Comment by cepprice** [comment]
@sonora In my case ffter tapping on Regensburg at zoom=7:
- with legacy app stalls for ~20s, then shows selection menu with tens of routes
- with OpenGL app quickly shows selection menu with few items (and no routes)

In both cases all routes are disabled.

As far as I understoon, you have lags with both legacy and OpenGL?

**Comment by vshcherb** [comment]
Ok, then probably won't fix ...

**Comment by sonora** [comment]
Quite sure it's a regression, though. I believe that
- this is likely due to mostly all the routes, which take a long time to collect
- and think that previously e.g. in the Browse map profile we never even had the routes tappable (only under their respective profile), but now they seem tappable in all profiles, and at all zoom levels?

**Comment by sonora** [comment]
This is a great fix, thank you, @cepprice !!

**Comment by yuriiurshuliak** [comment]
I attempted to reproduce the bug under the same conditions as last time. Presently, everything is functioning correctly, and the map moves as expected, regardless of the zoom level.

OsmAnd~ 4.7.0#1740m, released: 2024-03-04

https://github.com/osmandapp/OsmAnd/assets/127092082/959161a3-1643-4f10-8724-746d1e46bc1b
